week2/homework/src/index.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
+const {
+    add,
+    reset,
+    list,
+    remove,
+    update
+} = require("./IOutil");
 
-// TODO: Write the homework code in this file
+const TODO_FILE = 'todo.json';
+
+function main() {
+    const [, , cmd, ...args] = process.argv;
+
+    switch (cmd) {
+        case 'add':
+            add(TODO_FILE, "utf8", args);
+            break;
+
+        case 'list':
+            list(TODO_FILE);
+            break;
+
+        case 'reset':
+            reset(TODO_FILE);
+            break;
+
+        case 'remove':
+            remove(TODO_FILE, args);
+            break;
+
+        case 'update':
+            {
+                let index = args[0];
+                args.splice(0, 1);
+                update(TODO_FILE, index, args);
```

Comment:
Again, lack of consistency: your other functions just accepts the unchanged `args` as their second argument. You could define`index` and do the splicing inside of your `update` function so that your logic isn't spread across several places.

Reply:
Agreed, i should have put them inside the function, i don't know know why i did that lol